hyperspyui/plugins/alignzlp.py

```diff
@@ -24,4 +31,19 @@ def create_toolbars(self):
     def default(self):
         ui = self.ui
         s = ui.get_selected_signal()
-        s.align_zero_loss_peak()
+        logger.debug('Align zero loss peak of ' + str(s))
+        title = s.metadata.General.title
+
+        signal_list = [
+            sig for sig in ui.signals if sig.signal is not s and type(sig) == hyperspy.signals.EELSSpectrum]
```

Comment:
```suggestion
            sig for sig in ui.signals if (
                    sig.signal is not s and
                    isinstance(sig.signal, hyperspy.signals.EELSSpectrum))]
```
This is where the issue is, `sig.signal` should be checked instead of `sig`